tests/integration/test_shared_run_artifact_contract.py

```
@@ -0,0 +1,591 @@
+    executable = shutil.which("pwsh") or shutil.which("powershell")
+    if executable is None:
+        pytest.skip("PowerShell executable not available")
+    return executable
+
+
+def _ps_quote(value: str | Path) -> str:
+    return "'" + str(value).replace("'", "''") + "'"
+
+
+def _run_powershell_json(script: str) -> dict[str, object]:
+    completed = subprocess.run(
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)